v2/spanner-to-sourcedb/src/main/java/com/google/cloud/teleport/v2/templates/transforms/SourceWriterFn.java

```diff
@@ -0,0 +1,335 @@
+      ImmutableList<IndexColumn> keyColumns = table.primaryKeys();
+
+      for (IndexColumn keyColumn : keyColumns) {
+        Column key = table.column(keyColumn.name());
+        Type keyColType = key.type();
+        String keyColName = key.name(); // .toLowerCase();
+        switch (keyColType.getCode()) {
+          case BOOL:
+          case PG_BOOL:
+            pk.append(
+                ChangeEventTypeConvertor.toBoolean(
+                    keysJson, keyColName, /* requiredField= */ true));
+            break;
+          case INT64:
+          case PG_INT8:
+            pk.append(
+                ChangeEventTypeConvertor.toLong(keysJson, keyColName, /* requiredField= */ true));
+            break;
+          case FLOAT64:
+          case PG_FLOAT8:
+            pk.append(
+                ChangeEventTypeConvertor.toDouble(keysJson, keyColName, /* requiredField= */ true));
+            break;
+          case STRING:
+          case PG_VARCHAR:
+          case PG_TEXT:
+            pk.append(
+                ChangeEventTypeConvertor.toString(keysJson, keyColName, /* requiredField= */ true));
+            break;
+          case NUMERIC:
+          case PG_NUMERIC:
+            pk.append(
+                ChangeEventTypeConvertor.toNumericBigDecimal(
+                    keysJson, keyColName, /* requiredField= */ true));
+            break;
+          case JSON:
+          case PG_JSONB:
+            pk.append(
+                ChangeEventTypeConvertor.toString(keysJson, keyColName, /* requiredField= */ true));
+            break;
+          case BYTES:
+          case PG_BYTEA:
+            pk.append(
+                ChangeEventTypeConvertor.toByteArray(
+                    keysJson, keyColName, /* requiredField= */ true));
+            break;
+          case TIMESTAMP:
+          case PG_COMMIT_TIMESTAMP:
+          case PG_TIMESTAMPTZ:
+            pk.append(
+                ChangeEventTypeConvertor.toTimestamp(
+                    keysJson, keyColName, /* requiredField= */ true));
+            break;
+          case DATE:
+          case PG_DATE:
+            pk.append(
+                ChangeEventTypeConvertor.toDate(keysJson, keyColName, /* requiredField= */ true));
+            break;
+            // TODO(b/179070999) -  Add support for other data types.
+          default:
+            throw new IllegalArgumentException(
+                "Column name(" + keyColName + ") has unsupported column type(" + keyColType + ")");
+        }
+      }
+    } catch (Exception e) {
+      LOG.error("Failed to update last commit timestamp", e);
+    }
+    return pk.build();
+  }
+
+  private Mutation getShadowTableMutation(
+      String tableName,
+      String shadowTableName,
+      JsonNode keysJson,
+      com.google.cloud.Timestamp commitTimestamp) {
+    Mutation.WriteBuilder mutationBuilder =
+        Mutation.newInsertOrUpdateBuilder(shadowTableName)
+            .set("processed_commit_ts")
```

Comment:
Create a static final variable for "processed_commit_ts"

Reply:
Done